File: C#/50. Pow(x, n).cs
public class Solution {
    public double MyPow(double x, int n) {
        double res = 1;
        if(n == Int32.MaxValue){
            return (x == 1) ? 1 : (x == -1) ? -1 : 0;
        }
        if(n == Int32.MinValue){
            return (x == 1 || x == -1) ? 1 : 0;
        }
        if(n>0){
            for(int i = 0; i<n; i++){
                res*=x;
            }
            return res;
        }
        if(n<0){
            for(int i = 0; i>n; i--){
                res/=x;
            }
            return res;
        }
        return res;
    }
}

File: JavaScript/50. Pow(x, n).js
/**
 * @param {number} x
 * @param {number} n
 * @return {number}
 */
var myPow = function(x, n) {
    res = 1;
    if(n == 2147483647){
        return (x == 1) ? 1 : (x == -1) ? -1 : 0;
    }
    if(n == -2147483648){
        return (x == 1 || x == -1) ? 1 : 0;
    }
    if(n>0){
        for(var i = 0; i<n; i++){
            res*=x;
        }
        return res;
    }
    if(n<0){
        for(var i = 0; i>n; i--){
            res/=x;
        }
        return res;
    }
    return res;
};
